config/new_test.go

```diff
@@ -50,11 +50,6 @@ func TestNew_Invalid(t *testing.T) {
 		options       []HCPConfigOption
 		expectedError string
 	}{
-		{
-			name:          "missing credentials",
-			options:       []HCPConfigOption{},
-			expectedError: "the configuration is not valid: client credentials need to be provided",
-		},
```

Comment:
Question: I understand why this was taken out but what if the client only provides one of the two credentials? Will the browser login workflow just kick in anyways or do we assume the user wanted to try logging in with creds?

Reply:
That's a good point. I'll see about adding a check for when partial credentials are provided. But anytime both client credentials are missing, then we trigger browser login.